### DISCUSSION TITLE
👀 Maybe we need a detailed tutorial for add RDKit to PostgreSQL

### DISCUSSION THREAD

**Discussion by ShangChien** [rdkit/rdkit#6182] [Q&A] [answered]
Currently build postgresql with RDKit is not a easy thing.  Postgresql has multi-version: 9, 10 ,11, 12, 13,14 ,15. 
[Offical postgresql-rdkit introduction is still about Postgresql.9.*](https://www.rdkit.org/docs/Cartridge.html)
Really looking forward to having a complete new tutorial for more detail about how to add RDKit as extension to PostgreSQL.

Thank YOU 🎉 

Related Discussions: https://github.com/rdkit/rdkit/discussions/5158#discussion-3980750

**Comment by ShangChien** [comment]
```shell
(base) sanyue@cu06:~/webapp/dataBase/rdkit-Release_2022_09_5/Code/PgSQL/rdkit$ USE_PGXS=1 make installcheck
/usr/lib/postgresql/12/lib/pgxs/src/makefiles/../../src/test/regress/pg_regress --inputdir=./ --bindir='/usr/lib/postgresql/12/bin'    --dbname=contrib_regression rdkit-91 props btree molgist bfpgist-91 bfpgin sfpgist slfpgist fps reaction
(using postmaster on Unix socket, default port)
pg_regress: could not set core size: disallowed by hard limit
============== dropping database "contrib_regression" ==============
NOTICE:  database "contrib_regression" does not exist, skipping
DROP DATABASE
============== creating database "contrib_regression" ==============
CREATE DATABASE
ALTER DATABASE
============== running regression test queries        ==============
test rdkit-91                     ... ok         3277 ms
test props                        ... ok           73 ms
test btree                        ... ok          780 ms
test molgist                      ... ok         1887 ms
test bfpgist-91                   ... ok          123 ms
test bfpgin                       ... FAILED      232 ms
test sfpgist                      ... ok           95 ms
test slfpgist                     ... ok           98 ms
test fps                          ... ok           80 ms
test reaction                     ... ok         3370 ms

=======================
 1 of 10 tests failed.
=======================

The differences that caused some tests to fail can be viewed in the
file "/home/sanyue/webapp/dataBase/rdkit-Release_2022_09_5/Code/PgSQL/rdkit/regression.diffs".  A copy of the test summary that you see
above is saved in the file "/home/sanyue/webapp/dataBase/rdkit-Release_2022_09_5/Code/PgSQL/rdkit/regression.out".

make: *** [/usr/lib/postgresql/12/lib/pgxs/src/makefiles/pgxs.mk:420：installcheck] error 1

```
## I build rdkit from source(rdkit-20220905), then add it to postgresql-12 in ubuntu20.04. 
Anyone can help me point the meaning of  `rdkit-91 props btree molgist bfpgist-91 bfpgin sfpgist slfpgist fps reaction`?
What is the effect of  `test bfpgin  ... FAILED` ?
Thank you 👀

**Reply by ShangChien** [reply]
Dear @ptosco,  I'd appreciate it if you could guide me a bit  ( sorry to mention you)

**Reply by ptosco** [reply]
@ShangChien Updating the documentatio on the PostgreSQL cartridge build procedure is on my todo list - I hope I’ll manage to do it over the weekend.

**Reply by boglet** [reply]
Having failed three times, I too would love an installation guide for Windows. I followed the installation instructions but they stop working on step 5:
1) Install Anaconda - check!
2) Install RDkit - check!
3) Activate RDkit environment - check!
4) Install RDkit-Postgres using conda - check!
5) Run initdb - fail.  There is no initdb in [conda folder]/envs/my-rdkit-env/bin/

...so I am stuck.

Perhaps I am using the wrong approach and should build from source?  If so, which versions of PostgreSQL are supported (9.x is getting really old now!)?

**Comment by ptosco** [comment, marked as answer]
@ShangChien @boglet Yesterday I have updated the `README` file with up-to-date instructions.
All PostgreSQL versions from 9.x to current (15) are supported.
Those instructions apply to PostgreSQL installed using the Windows installer, which requires administrator privileges.
Building against `conda` definitely works and is actually easier as Windows administrator privileges are not required.

**Reply by ShangChien** [reply]
thank you

**Comment by boglet** [comment]
Thanks for the information.  

I tried again using Conda and finally succeeded. The documentation states that the **postgres** and **initdb** commands are in

` “[conda folder]/envs/my-rdkit-env/bin/” `

but it turns out that they are actually in 

`“[conda folder]/envs/my-rdkit-env/Library/bin/” `

Perhaps the documentation could be updated to reflect that? Maybe this is only the case for Windows installations?

I couldn't see a way of adding RDkit cartridge functionality to my existing Windows PostgreSQL installation.  Is there a simple way of doing this?

**Reply by ptosco** [reply]
@boglet Yes, Windows `conda` installations have an additional `Library` subdirectory that other platforms do not have.
There is updated installation information available here:
https://github.com/rdkit/rdkit/blob/master/Code/PgSQL/rdkit/README.md

**Comment by rccc** [comment]
For ubuntu/debian user:

apt-get install librdkit-dev librdkit1 python-rdkit postgresql-11-rdkit

su - postgres

psql -c 'CREATE EXTENSION rdkit' DB_NAME (should return "CREATE EXTENSION")

psql -d DB_NAME

select is_valid_smiles('CCC'); (should return 't' if  installation is ok)
